### PR TITLE
Improves running from source installation

### PR DESCRIPTION
**Pull request by tncowart**
This is a big bag of changes I made while helping a student on rotation try to get BEASTIE working on Duke's DCC environment (not using Docker). 

I changed a bunch of stuff related to importing files:
There were several issues with import statements, especially from misc_tools

These changes make several adjustments:
```
* Copies misc_tools into BEASTIE dir, removes git submodule
* Adds Capitalized files to misc_tools/__init__.py
* Makes imports relative, when appropriate
* sorts imports (using `isort` utility)
* removes unused imports
* removes python 2 "future" and "builtins" imports

Additionally, one bug is fixed:
* the logging file path code didn't join paths correctly

and I got rid of an unnecessary use of subprocess.call, as the python code could now
(due to fixing the imports) be called directly.
```

Fixed a couple of bugs:
```
Fixes a "referenced before assignment" bug

The data13 variable was only created on a single conditional path, but was
referenced after the conditional.

This moves the reference to the same path it's created.

And also fixes a little formatting.
```
and
```
Removes hardcoded values from annotate_LD_new.R

These hardcoded values were overwriting the arguments passed in from the command line.
```
and added some more information about loading R dependencies:
```
Add information about loading R packages to README

I added a little more information about loading the necessary R packages to the
README file.

I also adjust the Docker commands for loading the R dependencies a little.
```

### CODE REVIEW

File: BEASTIE/annotate_LD_new.R [hunk 1]
```diff
@@ -7,23 +7,14 @@ library("LDlinkR")
...
-prefix="HG00096_chr21"
-ancestry="EUR"
-infile="/Users/scarlett/Documents/Allen_lab/github/BEASTIE/BEASTIE_example/HG00096_chr21/output/s-0.5_a-0.05_sinCov0_totCov1_W1000K1000/HG00096_chr21_hetSNP_intersected_filtered.TEMP.tsv"
-outpath="/Users/scarlett/Documents/Allen_lab/github/BEASTIE/BEASTIE_example/HG00096_chr21/output/s-0.5_a-0.05_sinCov0_totCov1_W1000K1000/"
-mytoken="c313799c13c3"
-chr_start=1
-chr_end=22
-meta="/Users/scarlett/Documents/Allen_lab/github/BEASTIE/BEASTIE_example/HG00096_chr21/output/s-0.5_a-0.05_sinCov0_totCov1_W1000K1000/HG00096_chr21_meta.TEMP.tsv"
+chr_end=args[7]
+meta=args[8]
```
Comment (x811zou) [inline BEASTIE/annotate_LD_new.R:27]: For these lines, could you comment it out instead of removing it? it is for debugging purposes

**Review by x811zou — COMMENTED** [review]
@zackgomez has some thoughts about the set up of the misc_tools change, I will let him elaborate here

**Comment by zackgomez** [comment]
I made some similar changes to 'misc_tools' in my explorations but didn't go as far as copying them into the tree or any "real" solution.

What do you think about returning to git submodules against a fork of Bill's code that is updated with the relative imports?  This has two benefits- updates to Bill's code can be more easily imported in to the BEASTIE repo, and the relative import change could be contributed upstream to Bill's repo via PR.

**Comment by zackgomez** [comment]
I think the only meaningful changes I made in my branch that aren't covered or improved on by your branch are some work towards respecting default values when extracting values form the config in `BEASTIE/__init__.py`.  I think your changes to run avoid the extra process make it even easier to respect defaults.  These are the defaults from "BEASTIE.py"

image

I also added `scipy == 1.5.4` to `requirements.txt`

And finally I migrated the os.system calls to subprocess.run and passed options check=True, and stdout/stderr=PIPE.

**Comment by tncowart** [comment]
@zackgomez @x811zou Putting the defaults in the config values is a nice improvement. If you want to pull that out into a PR, it would be good to merge that in.

It's been a while, but I think I forked Bill's code for a couple of reasons.
1) Making the imports relative was a kind of structural change that I'm not sure is backwards compatible. I don't want to break other folks code.
2) I wanted all the python 2 stuff gone, but I'm not sure that Bill does.
3) I wasn't sure what a good module name for the installed code would be.
4) Bill's code doesn't have a `__init__.py` file, and so isn't a module that can be imported.

(3) isn't a big deal (I think I could have moved the submodule to the `BEASTIE/misc_tools` directory) but before we go back to using Bill's code as a submodule, we should resolve (1), (2), and (4) with Bill. My preference is to commit `misc_tools` as is then talk with Bill about (1), (2), and (4). If we get that resolved and checked in to his repo we can change `BEASTIE/misc_tools` to a submodule.

**Comment by zackgomez** [comment]
1 & 2) Totally agree. But a fork with these changes + submodule provides a clearer path for improvements from Bill's code to be imported into BEASTIE than a direct copy and modification.
4) I believe it has an empty `__init__.py` which is why you can import as `misc_tools.X` and use relative imports at all.

Python isn't my primary language so I don't have a strong feeling for project structure but agree with your assessment of (3).

So to be clear my suggestion is a forked copy of bill's code (maybe tncowart/ or x811zou/) as submodule under `BEASTIE/misc_tools`.  It doesn't seem like a large work amount of work, but will defer to you if you think it's much more expedient to go with the import approach.

I'll rebase my branch on master and submit a PR for the config changes when this PR is closed.

**Comment by tncowart** [comment]
@zackgomez I think your suggestion is reasonable. I've recreated the submodule in BEASTIE/misc_tools.

**Pull request merged**